### DISCUSSION TITLE
Add support for multiple matrix accounts via matrix-web and/or element-desktop

### DISCUSSION THREAD

**Discussion by stevleibelt** [element-hq/element-meta#1681] [Ideas]
This discussion is based on [issue/2320](https://github.com/vector-im/element-web/issues/2320) raised by [richvdh](https://github.com/richvdh) on 20.09.2016.
[t3chguy](https://github.com/t3chguy) suggested that this place here would be a better one to discuss this idea.

What is intended as a feature is to be able to manage and use multiple matrix accounts in all of the matrix front-ends.

Thanks to the success of matrix, it is common to have a private- and a work-account. Thanks to the success of remote working, it is common to use both accounts in parallel on the phone or on the desktop.

As pointed out by [paboum](https://github.com/paboum) on [April 3.](https://github.com/vector-im/element-web/issues/2320#issuecomment-1494629755), [Fluffy-Chat](https://gitlab.com/famedly/fluffychat/-/blob/main/CHANGELOG.md#v0410-2021-09-14) had added support for this feature on its 0.4.10 release.

Cheers,
Stev

**Comment by t3chguy** [comment]
My suggestion was to a specific commenter who wanted a per-space identity within a single account instead of multiple account support ftr

**Reply by genofire** [reply]
a nice idea for a feature. But that is no replacement for multiaccount support.

I have multiple for work and for private. My company like to have the control over there server, so a special space where i join with my privat account whould noz be possible.

**Reply by genofire** [reply]
Found option to rename in room `/myroomnick`. Spaces are also only rooms ... but there is no option in any UI (Element) for that. A rule ofer inheritage of nickname is unclear.

**Reply by t3chguy** [reply]
@genofire how would you suggest such inheritance to work for rooms which are in multiple spaces?

**Reply by genofire** [reply]
nope, i believe to define it manuelle everywhere is the cleanes option ...

So my suggestion would be to add real multi-account support

**Comment by genofire** [comment]
To remember the list of issues and Discussion, here an overview:
https://github.com/vector-im/roadmap/issues/11#issuecomment-1487242641

(just as link here, i like to improve it, if another one pop up)

PS: Hope we good an discussion about the how and bot about the if ... it is obviously that element need it.

**Reply by genofire** [reply]
found another discussion: https://github.com/vector-im/element-meta/discussions/1832

**Comment by BloodyIron** [comment]
Maybe instead of matrix-web or element-desktop we could have it AND? ;)

**Comment by ghost** [comment]
Just throwing it out there, I have a use case now where I have a need for two matrix accounts.
- A workplace issued account for a home server for work
- A personal account I use for a personal home server

It would be nice to access all my chats from one Element Web session. It also would be more beneficial on the iOS clients so my phone can access both.

**Comment by KeebCat** [comment]
Just in case this isn't already planned, once this feature is added, it would be nice if push notifications worked for all logged in accounts, not just the currently active one, if that's reasonably possible :)

**Comment by cadaniel** [comment]
I have a use case for this as well. 

I have disassociative identity disorder, or by what used to be known as multiple personality syndrome. 

I'd like to create an app where all my different alternative selves (alters) can not only chat with each other, but with friends and have them separated out by alter instead of user. Though multiple players can be up front at the same time.

**Reply by beedell-roke** [reply]
@cadaniel, does what you request differ from standard multiple account support?

**Comment by bew** [comment]
I was redirected from the closing of tracking issue https://github.com/element-hq/element-web/issues/2320#issuecomment-2547876077 .

I completely understand that it is not a priority at the moment and it can be hard to do reliability across all platforms.
By no means is it that I don't like your work, I'm grateful for all the work being done around Matrix it's really cool✨

.. But **this** is not cool, we're being redirected again and again to follow progress on this quite highly requested feature (700+ likes on OP of the previous issue).
👉 Is there a single ISSUE (not discussion) to subscribe to for when this will start being worked on seriously?

There is this roadmap ticket: https://github.com/element-hq/roadmap/issues/11
Is this a good tracking issue for this feature?

---
#### Aside, ISSUE please not discussion
Discussions are a huge pain to follow, because Github doesn't offer ability to subscribe to the status change (open/close) events, and we can _only_ subscribe to every single message of a discussion, which can generate a lot of notifications noise..
👉 Please avoid redirected people to a discussion 😬

**Reply by beedell-roke** [reply]
> https://github.com/element-hq/element-meta/discussions/1681#discussioncomment-11591812

@bew, I affirm what you've stated. I was previously subscribed solely to issue closure and reopening. I suggest you follow [`orgs/community/discussions/61069#discussion-5411152`](https://github.com/orgs/community/discussions/61069#discussion-5411152), which I have filed.

**Comment by alfredonodo** [comment]
Any news for this fundamental feature?
Thank you

**Reply by BloodyIron** [reply]
One might say it's... Elemental.
